src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Vault/SetAzureRmRecoveryServicesVaultProperty.cs

```diff
@@ -0,0 +1,61 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Management.RecoveryServices.Backup.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets.ServiceClientAdapterNS;
+using Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets.Models;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets
+{
+    /// <summary>
+    /// Used to set RecoveryServices Vault properties
+    /// </summary>
+    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RecoveryServicesVaultProperty", SupportsShouldProcess = true), OutputType(typeof(BackupResourceVaultConfigResource))]
+    public class SetAzureRmRecoveryServicesVaultProperty : RSBackupVaultCmdletBase
+    {
+        [Parameter(Mandatory = true, ValueFromPipeline = false)]
+        [ValidateNotNullOrEmpty]
+        public VaultSoftDeleteFeatureState SoftDeleteFeatureState { get; set; }
```

Comment:
It will be better to set the type of `SoftDeleteFeatureState` as string and  set the value range to `Enabled` and `Disabled`. You can find example [here](https://github.com/Azure/azure-powershell/blob/master/src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs#L111)

Reply:
@wyunchi-ms : I have seen the example but I am still not convinced/clear. I want to set the feature state with this command and hence I want to "enable/disable" it. Later when I get the value, I am expecting the feature to be either "Enabled/Disabled". May I know what is incorrect in this assumption?

Reply:
Both `Enable` and `Enabled` are fine. You can choose by youself. But I suggest to use `ValidateSet` to make sure user input the right value that you expect.

Reply:
The type of `SoftDeleteFeatureState` should be string and `VaultSoftDeleteFeatureState` is no need any more. You can use `ValidateSet` to ensure the user input the right value, and there is no need to add a new type for this.